src/m365/outlook/commands/mailbox/mailbox-settings-set.ts

```diff
@@ -77,6 +77,10 @@ class OutlookMailboxSettingsSetCommand extends GraphCommand {
         throw 'When running with application permissions either userId or userName is required, but not both';
       }
 
+      if (!(args.options.userId || args.options.userName)) {
```

Comment:
Let's do this specific check in the zod validation phase. Regardless of if the context is application permissions, only one option should be used.